bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/SncfBridgeHandler.java

```diff
@@ -0,0 +1,177 @@
+import org.openhab.binding.sncf.internal.dto.Passages;
+import org.openhab.binding.sncf.internal.dto.PlaceNearby;
+import org.openhab.binding.sncf.internal.dto.PlacesNearby;
+import org.openhab.binding.sncf.internal.dto.SncfAnswer;
+import org.openhab.binding.sncf.internal.dto.StopPoint;
+import org.openhab.binding.sncf.internal.dto.StopPoints;
+import org.openhab.core.cache.ExpiringCacheMap;
+import org.openhab.core.i18n.LocationProvider;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link SncfBridgeHandler} is handles connection and communication toward
+ * SNCF API
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class SncfBridgeHandler extends BaseBridgeHandler {
+    public static final String JSON_CONTENT_TYPE = "application/json";
+
+    public static final String SERVICE_URL = "https://api.sncf.com/v1/coverage/sncf/";
+
+    private final Logger logger = LoggerFactory.getLogger(SncfBridgeHandler.class);
+    private final LocationProvider locationProvider;
+    private final ExpiringCacheMap<String, String> cache = new ExpiringCacheMap<>(Duration.ofMinutes(1));
+    private final HttpClient httpClient;
+
+    private final Gson gson;
+    private @NonNullByDefault({}) String apiId;
+
+    public SncfBridgeHandler(Bridge bridge, Gson gson, LocationProvider locationProvider, HttpClient httpClient) {
+        super(bridge);
+        this.locationProvider = locationProvider;
+        this.httpClient = httpClient;
+        this.gson = gson;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing SNCF API bridge handler.");
+        apiId = (String) getConfig().get("apiID");
+        if (apiId != null && apiId.length() != 0) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/null-or-empty-api-key");
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("SNCF API Bridge is read-only and does not handle commands");
+    }
+
+    private <T extends SncfAnswer> T getResponseFromCache(String url, Class<T> objectClass) throws SncfException {
+        String answer = cache.get(url);
+        if (answer == null) {
+            final String newAnswer = getResponse(url);
+            cache.put(url, () -> newAnswer);
+            answer = newAnswer;
+        }
+        @Nullable
+        T response = gson.fromJson(answer, objectClass);
```

Comment:
Are you sure that answer cannot be null at this place ? 
Maybe you should catch the JSON exception.

Reply:
Changed.